doc/threading_notes.md

```diff
@@ -0,0 +1,49 @@
+# Threading model and optimizing for resource constrained devices
+
+## IoTHub Device Client and the \_LL\_ (lower layer) API
+
+### The lower layer
+
+There are two distinct but closely related sets of APIs for the IoTHub client.
+Applications that want to run on a single thread should use the \_LL\_ (aka lower level) API.  Sometimes the hardware/OS can only support one thread at a time or else spinning additional threads is expensive.  Applications on more capable hardware may also use the \_LL\_ layer for greater control over scheduling.
+
+Calls to the IoThub API's ending in Async do not cause the operation to execute immediately.  Instead, the work - both for sending data across the network and receiving data - is queued.  The application must manually schedule the work by calling `IoTHubDeviceClient_LL_DoWork` (when using a device client).
```

Comment:
By "when using a device client" you mean instead of using module client? The average reader might not have that distinction in mind.

Reply:
Linked to section where this is spelled out in more detail after this bit.